app/server/lib/DocApi.ts

```diff
@@ -412,6 +412,24 @@ export class DocWorkerApi {
       }
 
       if (tableId !== undefined) {
+        if (columnIds) {
+          if (tableId !== currentTableId && currentTableId) {
+            // if the tableId changed, we need to reset the columnIds
+            fields.columnRefList = [GristObjCode.List];
+          } else {
+            if (!tableId) {
+              throw new ApiError(`Cannot find columns "${columnIds}" because table is not known`, 404);
+            }
+            // columnIds have to be of shape "columnId; columnId; columnId"
+            fields.columnRefList = [GristObjCode.List, ...columnIds.split(";")
```

Comment:
So here, we are in the back-end, defining the behavior of a REST API endpoint. Sending `;`-separated columns seems pretty inconsistent with how lists are handled elsewhere in the API. I can understand doing that in the minimalist front-end UI, just because of lack of time to do something pretty (the same trade-off we made for isReadyColumn), but is there a way to isolate that to the front end? Once the back-end does something, changing it is hard since someone may rely on it.

It will be sad if we eventually add a pretty UI for this and it ends up having the convert a nice structured list of column ids into a semicolon separated string to send to the back end.

Reply:
`columnIds` is a little vague, and a bit inconsistent with `isReadyColumn` which doesn't specify the `Id` part. Maybe something like `watchedColumns`?

Reply:
Also, if column ids are stored as strings, would need some thinking about updating them when columns are renamed.

Reply:
This remark is a very good point, I isolated this logical in the frontend. 
The `watchedColIds` is stored in the db as `watchedColRefList` so if the column is renamed it's works well